### ISSUE TITLE
Adding a dependency on PyTorch

### ISSUE THREAD

**Issue by sogartar** [iree-org/iree-test-suites#57]
Is it acceptable that some tests have a dependency on PyTorch? There is the problem that NumPy does not have support for bfloat16.
Now I mostly want this to convert a blfoat16 IREE buffer view to float32.
You may want to operate on bfloat16 tensors without conversion.

**Comment by ScottTodd** [comment]
Yes, though I would prefer for such tests to be scoped to a `pytorch_` folder, like how onnx tests are in the `onnx_` folders. See also https://github.com/iree-org/iree-test-suites/issues/17 for setting up tests, with some sample code ideas.

If you just want utilities for working with data types, I'd prefer to _not_ take a dependency on a full ML framework just for that.

**Comment by ScottTodd** [comment]
Also depending on what you want, you can follow a similar pattern to the onnx op tests. Those depend on onnx for test case generation, writing out `.bin` files with inputs and expected outputs so the generated tests can be run using standard IREE tools with no framework dependency.

* Docs: https://github.com/iree-org/iree-test-suites/tree/main/onnx_ops
* Code: https://github.com/iree-org/iree-test-suites/blob/main/onnx_ops/import_onnx_tests.py
* Sample test: https://github.com/iree-org/iree-test-suites/tree/main/onnx_ops/onnx/node/generated/test_abs

**Comment by rsuderman** [comment]
Its not firm but I would recommend sticking to numpy if possible. We can pretty easily convert `bfloat16` to `float32` just by bit extending and shifting. The main justification is that the CI time is dominated by pytorch installation if we include it.

**Comment by sogartar** [comment]
@ScottTodd, thank you for the suggestion. The comparison function I want to use is the cosine similarity per token of the returned features from a text encoder. An element-wise closeness comparison would not be good enough.

The other option is to have a second IREE module to compute the cosine similarity that would return float32. This does not seem right to me to use an IREE module to test IREE's execution of a module.

> We can pretty easily convert bfloat16 to float32 just by bit extending and shifting.

@rsuderman, I assume we can achieve this with NumPy by treating the value as int. I will write this function then.

**Comment by sogartar** [comment]
I opened an [issue](https://github.com/iree-org/iree/issues/19471) regarding the possibility of interop with dtypes from `ml_dtypes`.
